controller/lib/hal/hal.h

```diff
@@ -130,7 +130,12 @@ class HalApi {
   void init();
 
   // Current time, strongly typed.  Prefer this over millis().
-  Time now() { return millisSinceStartup(millis()); }
+  Time now() {
+    // TODO: millis() is a uint32_t, which means it rolls over after about a
+    // month.  Use a routine that doesn't roll over, or detect and handle
+    // rollover.
```

Comment:
Or have an alarm that says "time to restart the device after this patient is off the ventilator" (modulo whatever the requirements say about how long the ventilator must be able to stay turned on and operational).

Reply:
lol I guess.  It should be easy to fix the overflow in code.  We have a watchdog timer that ensures we make one trip through the main loop every 120ms.  In order to catch overflow (time goes backwards), we have to query `millis()` at least once every 2^32/2 milliseconds = 0.8 months.  So we have a little wiggle room.